### ISSUE TITLE
[Mailer] sendmail transport not working on Windows

### ISSUE THREAD

**Issue by Seb33300** [symfony/symfony#36842]
**Symfony version(s) affected**: 5.0.8

**Description**  
I am upgrading my application to Symfony 5 and moving from SwiftMailer to Symfony Mailer.

I would like to send emails using sendmail:
```
MAILER_DSN=sendmail+smtp://default
```

This is working well on Linux but not on Windows.
I have properly configured my php.ini by setting the `sendmail_path` option but it looks like Symfony does not take it in account.

When sending an email, this error is returned:
> Process could not be started: Le chemin d'accès spécifié est introuvable.
> in C:\dev\vendor\symfony\mailer\Transport\Smtp\Stream\ProcessStream.php (line 44) 

While trying to debug, I found that `$this->command` contains `/usr/sbin/sendmail -bs` which looks to be the default linux path for sendmail.

**Comment by fabpot** [comment]
Were you using sendmail on Windows with Swiftmailer as well? If yes, how did you configure it?

**Comment by Seb33300** [comment]
I think the way I presented this was confusing but it was not working with SwiftMailer too.

**Comment by nescim** [comment]
### Found the problem:

I had a look at `SendmailTransportFactory`, and found these three lines (version 4.4.10, lines 23-25):
```php
        if ('sendmail+smtp' === $dsn->getScheme() || 'sendmail' === $dsn->getScheme()) {
            return new SendmailTransport(null, $this->dispatcher, $this->logger);
        }
```
The issue seems to be that the factory is disregarding the path given in the DSN (never calls `getHost()`), and just passes `null` as the command to `SendmailTransport`'s constructor.
On its end, `SendmailTransport` defines the command property with a default value:
```php
    private $command = '/usr/sbin/sendmail -bs';
```
and does not overwrite it if the passed parameter is `null`.

`php.ini` never enters the picture.

### How to fix it
A one-line fix may be to put `$dns->getHost()` in place of null in the factory? I'll try that later but I'm afraid there would be some additional work needed (do spaces need to be escaped?).
A nice feature could be to use `php.ini` configuration if the host (command) is "default".

Personally, I doubt that having the default command hardcoded in `SendmailTransport` is a good idea but it can be handy and altering that would likely be a BC break.

**Comment by fabpot** [comment]
This has been fixed now in 5.2 where we have a new NativeTransport that reads configuration from php.ini. See #36131
